Buildings/Controls/OBC/CDL/Continuous/NormalizedDelay.mo

```diff
@@ -0,0 +1,64 @@
+within Buildings.Controls.OBC.CDL.Continuous;
+block NormalizedDelay "Calculate a time delay based on a relay experiment"
+  parameter Real gamma=3 "Asymmetry level of a relay tuner";
+  Interfaces.RealInput dtON "Half-period length for the higher value"
+    annotation (Placement(transformation(extent={{-140,40},{-100,80}}), iconTransformation(extent={{-140,40},{-100,80}})));
+  Interfaces.RealInput dtOFF "Half-period length of the lower value"
+    annotation (Placement(transformation(extent={{-140,-80},{-100,-40}}), iconTransformation(extent={{-140,-80},{-100,-40}})));
+  Interfaces.RealOutput y "Normalized time delay"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}})));
+  Max max annotation (Placement(transformation(extent={{-62,40},{-42,60}})));
+  Min min annotation (Placement(transformation(extent={{-60,-64},{-40,-44}})));
+  Divide rho "Half-period ratio" annotation (Placement(transformation(extent={{-16,-10},{4,10}})));
+  Modelica.Blocks.Sources.RealExpression zeroCheck(y=noEvent(if min.y > 0 then min.y else 1))
+    annotation (Placement(transformation(extent={{-58,-30},{-38,-10}})));
+equation
+  assert(abs(gamma-1)>0.01, "The relay signal should be asymmetric");
+//  assert(rho.y>1, "The half-period ratio should be larger than 1");
+//  assert(y>1 or y<0, "Invalid value for the normalized time delay");
+  y=(gamma-rho.y)/(gamma-1)/(0.35*rho.y+0.65);
+  connect(max.u1, dtON) annotation (Line(points={{-64,56},{-80,56},{-80,60},{-120,60}}, color={0,0,127}));
```

Comment:
Don't mix graphical modeling with textual, see https://simulationresearch.lbl.gov/modelica/userGuide/development.html#equations-and-algorithms

Reply:
Only graphical modeling (with components from CDL) is used in this package